contracts/PatriciaTree/Data.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity ^0.4.16;
+pragma experimental "v0.5.0";
+pragma experimental "ABIEncoderV2";
+
+import {Bits} from "./Bits.sol";
+
+
+/*
+ * Data structures and utilities used in the Patricia Tree.
+ *
+ * More info at: https://github.com/chriseth/patricia-trie
+ */
+library Data {
+
+  struct Label {
+    bytes32 data;
+    uint length;
+  }
+
+  struct Edge {
+    bytes32 node;
+    Label label;
+  }
+
+  struct Node {
+    Edge[2] children;
+  }
+
+  struct Tree {
+    bytes32 root;
+    Data.Edge rootEdge;
+    mapping(bytes32 => Data.Node) nodes;
+  }
+
+  // Returns a label containing the longest common prefix of `self` and `label`,
+  // and a label consisting of the remaining part of `label`.
+  function splitCommonPrefix(Label memory self, Label memory other) internal pure returns (
+    Label memory prefix,
+    Label memory labelSuffix
+  )
+  {
+    return splitAt(self, commonPrefix(self, other));
+  }
+
+  // Splits the label at the given position and returns prefix and suffix,
+  // i.e. 'prefix.length == pos' and 'prefix.data . suffix.data == l.data'.
+  function splitAt(Label memory self, uint pos) internal pure returns (Label memory prefix, Label memory suffix) {
+    assert(pos <= self.length && pos <= 256);
+    prefix.length = pos;
+    if (pos == 0) {
+      prefix.data = bytes32(0);
+    } else {
+      prefix.data = bytes32(uint(self.data) & ~uint(1) << 255 - pos);
+    }
+    suffix.length = self.length - pos;
+    suffix.data = self.data << pos;
+  }
+
+  // Returns the length of the longest common prefix of the two labels.
+  /*
+  function commonPrefix(Label memory self, Label memory other) internal pure returns (uint prefix) {
+    uint length = self.length < other.length ? self.length : other.length;
+    // TODO: This could actually use a "highestBitSet" helper
+    uint diff = uint(self.data ^ other.data);
+    uint mask = uint(1) << 255;
+    for (; prefix < length; prefix++) {
+      if ((mask & diff) != 0) {
+        break;
+      }
+      diff += diff;
+    }
+  }
+  */
+
+  function commonPrefix(Label memory self, Label memory other) internal pure returns (uint prefix) {
```

Comment:
Maybe submit a PR to Chris's repo for this todo fix?

Reply:
Have I managed to butcher a rebase such that this isn't attached to what it should be?

Reply:
Oh, that wasn't me. This is a slightly updated version of the Patricia Tree taken from the [solidity-examples](https://github.com/ethereum/solidity-examples) repo.